src/Relude/Foldable/Fold.hs

```diff
@@ -194,12 +193,11 @@ Just True
 >>> orM [Nothing, Just True]
 Nothing
 -}
-orM :: (Foldable f, Monad m) => f (m Bool) -> m Bool
+orM :: forall f m . (Foldable f, Monad m) => f (m Bool) -> m Bool
 orM = foldr go (pure False)
   where
-    go p acc = do
-        q <- p
-        if q then pure True else acc
+    go :: m Bool -> m Bool -> m Bool
+    go p = ifM p (pure True)
```

Comment:
One more point goes to the benefits of types and explicit type signatures. After seeing the type signature and the implementation clearly, I noticed that we can reuse `(&&^)` and `(||^)` operators from `Relude.Bool.Guard`. So the two functions from this PR can be implemented even simpler:

```haskell
andM = foldr (&&^) (pure True)
orM  = foldr (||^) (pure False)